[Actions] Zoom new Actions

vellames-turing opened the pull request:
New Actions

- [x] Create Meeting | Create a new Zoom meeting
- [x] Create Meeting Registrant | Create a new meeting registrant
- [x] Create Webinar Registrant | Create a new webinar registrant
- [x] Get Meeting | Retrieve the details of a meeting
- [x] List Meetings | List all meetings, optionally by category
- [x] Update Meeting | Update the details of a meeting
- [x] End Meeting | End a meeting for a user
- [x] Delete Meeting | Remove a meeting by ID
- [x] List Meeting Registrants | List all registrants of a meeting
- [x] Update Meeting Registrant Status | Update registrant status for a meeting
- [x] Create Webinar | Create a webinar for a user
- [x] Get Webinar | Retrieve the details of a webinar
- [x] List Webinars | List all webinars of a user
- [x] Update Webinar | Update the details of a webinar
- [x] Delete Webinar | Remove a webinar for a user
- [x] List Webinar Registrants | List all registrants of a webinar
- [x] Update Webinar Registrant Status | Update registrant status for a webinar
- [x] Add Webinar Panelists | Register multiple panelists for a webinar
- [x] Remove Webinar Panelist | Remove a single panelist from a webinar
- [x] List Cloud Recordings | Search cloud recordings from a meeting or webinar
- [x] Delete a Cloud Recording | Remove a recording from a meeting or webinar

From CommonJS to ESM

- [x] app
- [x] (Source) Custom Event
- [x] (Source) Meeting Created
- [x] (Source) Meeting Deleted
- [x] (Source) Meeting Ended
- [x] (Source) Meeting Started
- [x] (Source) Meeting Updated
- [x] (Source) Recording Completed
- [x] (Source) Webinar Created
- [x] (Source) Webinar Deleted
- [x] (Source) Webinar Ended
- [x] (Source) Webinar Started
- [x] (Source) Webinar Updated

From CommonJS to ESM

- [x] app
- [x] (Source) Account Created
- [x] (Source) Account Settings Updated
- [x] (Source) Account Updated
- [x] (Source) Custom Events
- [x] (Source) Meeting Created
- [x] (Source) Meeting Deleted
- [x] (Source) Meeting Ended
- [x] (Source) Meeting Started
- [x] (Source) Meeting Updated
- [x] (Source) Recording Completed
- [x] (Source) User Activated
- [x] (Source) User Created
- [x] (Source) User Deactivated
- [x] (Source) User Deleted
- [x] (Source) User Invitation accepted
- [x] (Source) User Updated
- [x] (Source) Webinar Changes to Panelists
- [x] (Source) Webinar Created
- [x] (Source) Webinar Deleted
- [x] (Source) Webinar Ended
- [x] (Source) Webinar Started
- [x] (Source) Webinar Updated

Bugfix and Improvements
- [x] (App) Sometimes webinars are duplicated on the list
- [x] (App) Update every source using `webinars` variable to use object with `label`, `value` as value
- [x] Add structured values sanitization in non-structured mode

dylburger commented:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

dannyroosevelt commented:
> Hey @dannyroosevelt, @michelle0927 . About the **Download Cloud Recording** action. The **List Cloud Recordings** already returns the URL to download the records. I can create a method to just return an array of URLs, but, in my view, it looks redundant because the user already can use the list to get it. I am removing this action from the TODO list for now. Please let me know if you disagree or if I am missing something

That makes sense @vellames-turing, thanks for the heads up!

dannyroosevelt commented:
@vellames-turing as soon as you're ready, you should submit what you've got for review and we'll skip the remaining actions!

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+    cloudRecording: {
+      type: "string",
+      label: "Recording",
+      description: "The ID of the cloud recording. Please use a valid object with `meetingId` and `value` with \"Structured Mode\" disabled. `value` means the recording ID",
```
Can you add like an example something like
```suggestion
      description: "The ID of the cloud recording. Please use a valid object with `meetingId` and `value` with \"Structured Mode\" disabled. `value` means the recording ID (Eg. `{{ { meetingId:123, value:123 } }}`)",
```

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+              label,
+              value: {
+                meetingId: meeting.id,
+                label,
```
I think is not necessary to set the label inside `value` right?

vellames-turing replied:
Please check my comment in `zoom_admin.app.mjs` :)

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+          options: data?.meetings.map((meeting) => ({
+            label: meeting.topic,
+            value: {
+              label: meeting.topic,
```
I think is not necessary to assign the label again here for `$summary` porpuses since the summary can be more generic without the need of specific `ids` or `labels`. According to the [docs](https://pipedream.com/docs/components/guidelines/#use-summary-to-summarize-what-happened) the structure should be simple right? the same commet goes for the rest of `async options` with the same aproach where you are including the `label` in the `value`.

vellames-turing replied:
I disagree with it. According to the docs: "_Whenever possible, use names and titles instead of IDs_". The better way to do it is sending an object with the value and the name. Please check this [merged PR](https://github.com/PipedreamHQ/pipedream/pull/1762) using this same approach. If you still disagreeing, please let me know. Maybe we can move this discussion to Slack in order to found the best approach :)

jcortes replied:
Ok let me ask this on Slack because if that is true, it should be like an standard for the rest of PRs rigth?

vellames-turing replied:
Yeah, let's try to get a more detailed answer from @dannyroosevelt  or @dylburger and so we can go ahead with the result of our slack thread. :)

vellames-turing replied:
In my view, it should be optional, the dev could have the possibility to follow a more detailed approach like this, or use simple messages.

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+    registrants: {
+      type: "string[]",
+      label: "Registrants",
+      description: "The meeting registrant. If you disable the \"Structured Mode\", please provide an array of objects with `id` and `email`",
```
Can you put like a practical example for users to just be able to copy paste, something like
```suggestion
      description: "The meeting registrant. If you disable the \"Structured Mode\", please provide an array of objects with `id` and `email`. (Eg. `{{ [ { id: 123, email: \"123@mail.com\" } ] }}`)",
```

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+      }) {
+        let registrants = [];
+        const promises = [];
+        const statuses = [
+          "pending",
+          "approved",
+          "denied",
+        ];
```
I would suggest setting these kind of values as constants maybe in a `constants.mjs` file

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+          "denied",
+        ];
+
+        statuses.forEach((status) => {
+          promises.push(this.listMeetingRegistrants(
+            get(meeting, "value", meeting),
+            occurrenceId,
+            page + 1,
+            status,
+            isWebinar,
+          ));
+        });
```
```suggestion
        const promises = statuses.map((status) => 
          this.listMeetingRegistrants(
            get(meeting, "value", meeting),
            occurrenceId,
            page + 1,
            status,
            isWebinar,
          ));
```

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+        });
+
+        const registrantsPromisesResult = await Promise.all(promises);
+        registrantsPromisesResult.forEach((result) => {
+          registrants = registrants.concat(result);
+        });
```
Optionally you can also use `lodash.flatten` here
```suggestion
        const registrants = _.flatten(registrantsPromisesResult);
```

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+
+        const {
+          webinars,
+          next_page_token,
```
```suggestion
          next_page_token: nextPageToken,
```

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+      const { path } = opts;
+      delete opts.path;
+      // eslint-disable-next-line multiline-ternary
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
```
I think is not necessary to have this logic about having or not the `/` character in the path since we (developers) are controlling the paths in the requests right? So it would assume that we are starting all the paths like `/my/path` as an example

vellames-turing replied:
Yeah, I will rebuild the http handler

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+        headers: this._getHeaders(),
+      };
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers["Accept"] = "application/json";
+      opts.headers["Content-Type"] = "application/json";
+      opts.headers["Authorization"] = `Bearer ${this.$auth.oauth_access_token}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      // eslint-disable-next-line multiline-ternary
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return await axios(opts);
+    },
```
Can you refactor this a little bit more so we can
1. Use [pipedream axios](https://pipedream.com/docs/pipedream-axios/#using-pipedream-platform-axios-in-component-actions)
2. Reuse the headers function `_getHeaders` to set necessary headers
3. Centralize all `axios` calls making just use of `_makeRequest` so you don't have to import `axios` on  each isolated component

vellames-turing replied:
@jcortes As far as I know. It is not possible to use pipedream Axios here because I haven't access to `$` when I am filling the `options`. The `$` variable is only accessible through the `run` function that happens after the `options`. If you have some solution. Please let me know

jcortes replied:
What I normally do is to fallback to `this` if `$` is not present e.g.
```javascript
return await axios($ || this, config);
```

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return await axios(opts);
+    },
+    sanitizedArray(value) {
```
In my opinion this function looks more like a utility function right?

vellames-turing replied:
Yeah, what's your suggestion, to move it to an `utils` file?

jcortes replied:
Sure that would do it

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+      throw new Error(`${value} is not an array or an array-like`);
+    },
+    async listMeetings(nextPageToken) {
+      const { data } = await this._makeRequest({
+        path: "/users/me/meetings",
+        params: {
+          page_size: 100,
+          next_page_token: nextPageToken,
+        },
+      });
+
+      return data;
```
If you use pipedream's axios you can change this code as follows
```javascript
      return this._makeRequest({
        path: "/users/me/meetings",
        params: {
          page_size: 100,
          next_page_token: nextPageToken,
        },
      });
```

vellames-turing replied:
Yeah, I will rebuild the http handler

vellames-turing replied:
@jcortes As far as I know. It is not possible to use pipedream Axios here because I haven't access to $ when I am filling the options. The $ variable is only accessible through the run function that happens after the options. If you have some solution. Please let me know

jcortes replied:
@vellames-turing please see my comment [above](https://github.com/PipedreamHQ/pipedream/pull/1866#discussion_r758523787)

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+
+        return get(res, "data.registrants", []);
+      } catch (err) {
+        console.error(err);
```
I guess we can remove all `console.log`s and `console.error`s from code

vellames-turing replied:
Yeah, I will rebuild the http handler

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,374 @@
+        },
+      });
+
+      if (pageNumber > get(res, ("data.page_count"))) {
+        return [];
+      }
+
+      return get(res, "data");
```
I my opinion I would say is not ideal to deal with data logic in these functions. For me they just need to do one job and that is to return the response right from the api call using in this case `axios` lib.

jcortes reviewed (changes requested):
It's looking good so far. Since this is a huge PR this will be my first round of comments/suggestions on the app file.

vellames-turing commented:
Hey @jcortes, thanks for the review so far, I updated almost everything you asked. Please take a look in some of my resposes. Thank you!

jcortes reviewed (commented):
Hi @vellames-turing please take a look at [this comment](https://github.com/PipedreamHQ/pipedream/pull/1866#discussion_r758523787) and let me know if that works

vellames-turing commented:
Hey @jcortes . In this case, I usually use the `getAxiosParams` as a general builder for requests and I use the default `axios` inside de `app` since we dont have access to `$` during the prop options.I already have some approved PRs with this approach, It sounds good for you?

jcortes commented on components/zoom_admin/actions/add-meeting-registrant/add-meeting-registrant.mjs:
```diff
@@ -0,0 +1,198 @@
+      label: "Purchasing Time Frame",
+      description: "This field can be included to gauge interest of webinar attendees towards buying your product or service.",
+      optional: true,
+      options: [
+        "Within a month",
+        "1-3 months",
+        "4-6 months",
+        "More than 6 months",
+        "No timeframe",
+      ],
```
I would set all these variables as constants in a `constants.mjs` file, same goes for the rest for options are there

jcortes commented on components/zoom_admin/utils.mjs:
```diff
@@ -0,0 +1,22 @@
+    return value.map((item) => get(item, "value", item));
+  }
+
+  // If is string, try to convert it in an array
+  if (isString(value)) {
+    // Return an empty array if string is empty
+    if (isEmpty(value)) {
+      return [];
+    }
+
+    return value.replace(/["'[\]\s]+/g, "").split(",");
+  }
+
+  throw new Error(`${value} is not an array or an array-like`);
```
I would say at this point you can remove one nested `if` by doing it like this
```javascript
  if (!isString(value)) {
    throw new Error(`${value} is not an array-like string`);
  }
  
  // Return an empty array if string is empty
  if (isEmpty(value)) {
    return [];
  }

  // It is string, try to convert it in an array
  return value.replace(/["'[\]\s]+/g, "").split(",");
```

jcortes commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,333 @@
+        "user-agent": "@PipedreamHQ/pipedream v0.1",
+      };
+    },
+    _getAxiosParams(opts = {}) {
+      return {
+        ...opts,
+        url: this._apiUrl() + opts.path,
+        headers: this._getHeaders(),
+      };
+    },
```
as a suggestion you can also omit path variable from axios config properties like this
```javascript
    _getAxiosParams(opts = {}) {
      const {
        path,
        ...otherOpts,
      } = opts;
      return {
        ...otherOpts,
        url: this._apiUrl() + path,
        headers: this._getHeaders(),
      };
    },
```

jcortes commented on components/zoom_admin/zoom_languages.mjs:
```diff
@@ -0,0 +1,58 @@
+/*
+  Source: https://marketplace.zoom.us/docs/api-reference/zoom-api/meetings/meetingregistrantcreate
+          https://www.andiamo.co.uk/resources/iso-language-codes/
+          http://www.mathguide.de/info/tools/languagecode.html
+          https://en.wikipedia.org/wiki/Zh-TW
+          http://www.lingoes.net/en/translator/langcode.htm
+*/
```
```suggestion
/**
 * Source:
 * https://marketplace.zoom.us/docs/api-reference/zoom-api/meetings/meetingregistrantcreate
 * https://www.andiamo.co.uk/resources/iso-language-codes/
 * http://www.mathguide.de/info/tools/languagecode.html
 * https://en.wikipedia.org/wiki/Zh-TW
 * http://www.lingoes.net/en/translator/langcode.htm
 */
```

jcortes commented on components/zoom_admin/zoom_tzs.mjs:
```diff
@@ -0,0 +1,553 @@
+/*
+  Source: https://marketplace.zoom.us/docs/api-reference/other-references/abbreviation-lists#timezones
+  Create a simple crawler if you want to get the up to date timezones
+  1. Add "timezones-zoom" as the timezone table ID
+  2. Run the following code on your terminal https://pastebin.com/raw/npG4G0cU
+  4. Beatify your console.log (e.g. http://jsonviewer.stack.hu/)
+  3. Copy and paste your JSON after the "export default"
+  4. On the project root, execute "npx eslint --fix components/zoom_admin/zoom_tz.mjs" to fix
+  any ESLINT error.
+*/
```
```suggestion
/**
 * Source: https://marketplace.zoom.us/docs/api-reference/other-references/abbreviation-lists#timezones
 * Create a simple crawler if you want to get the up to date timezones
 * 1. Add "timezones-zoom" as the timezone table ID
 * 2. Run the following code on your terminal https://pastebin.com/raw/npG4G0cU
 * 4. Beatify your console.log (e.g. http://jsonviewer.stack.hu/)
 * 3. Copy and paste your JSON after the "export default"
 * 4. On the project root, execute "npx eslint --fix components/zoom_admin/zoom_tz.mjs" to fix
 * any ESLINT error.
*/
```

jcortes reviewed (approved):
I've just added final comments/suggestions. It looks great to me!

dannyroosevelt commented:
Partial QA done (still WIP):

Actions
- [x]  Create Meeting | Create a new Zoom meeting
- [x]  ~~Create~~ Add Meeting Registrant | ~~Create~~ Add a new meeting registrant
- [x]  ~~Create~~ Add Webinar Registrant | ~~Create~~ Add new webinar registrant
- [x]  Get Meeting | Retrieve the details of a meeting
- [x]  List Meetings | List all meetings, optionally by category
- [x]  Update Meeting | Update the details of a meeting
- [x]  Create Webinar | Create a webinar for a user

Remaining actions:
- [ ]  End Meeting | End a meeting for a user
- [ ]  Delete Meeting | Remove a meeting by ID
- [ ]  List Meeting Registrants | List all registrants of a meeting
- [ ]  Update Meeting Registrant Status | Update registrant status for a meeting 
- [ ]  Get Webinar | Retrieve the details of a webinar
- [ ]  List Webinars | List all webinars of a user
- [ ]  Update Webinar | Update the details of a webinar
- [ ]  Delete Webinar | Remove a webinar for a user
- [ ]  List Webinar Registrants | List all registrants of a webinar
- [ ]  Update Webinar Registrant Status | Update registrant status for a webinar
- [ ]  Add Webinar Panelists | Register multiple panelists for a webinar
- [ ]  Remove Webinar Panelist | Remove a single panelist from a webinar
- [ ]  List Cloud Recordings | Search cloud recordings from a meeting or webinar
- [ ]  Delete a Cloud Recording | Remove a recording from a meeting or webinar

And Sources
- [ ]  Account Created
- [ ]  Account Settings Updated
- [ ]  Account Updated
- [ ]  Custom Events
- [ ]  Meeting Created
- [ ]  Meeting Deleted
- [ ]  Meeting Ended
- [ ]  Meeting Started
- [ ]  Meeting Updated
- [ ]  Recording Completed
- [ ]  User Activated
- [ ]  User Created
- [ ]  User Deactivated
- [ ]  User Deleted
- [ ]  User Invitation accepted
- [ ]  User Updated
- [ ]  Webinar Changes to Panelists
- [ ]  Webinar Created
- [ ]  Webinar Deleted
- [ ]  Webinar Ended
- [ ]  Webinar Started
- [ ]  Webinar Updated

dannyroosevelt commented:
@vellames-turing FYI I'm removing (commenting out) a lot of optional props. In general we should aim to only pull in 3-4 optional props for a given component.

Also, it looks like a lot of the props are duplicated across specific components instead of having them live in the app file. Do you think it makes sense to consolidate them into the app file?

vellames-turing commented:
> @vellames-turing FYI I'm removing (commenting out) a lot of optional props. In general we should aim to only pull in 3-4 optional props for a given component.
> 
> Also, it looks like a lot of the props are duplicated across specific components instead of having them live in the app file. Do you think it makes sense to consolidate them into the app file?

Oh, ok, got it. Please let me know when you finish it, So I can check for duplicated props and consolidate it in `app` file

dannyroosevelt commented on components/zoom_admin/zoom_admin.app.mjs:
```diff
@@ -0,0 +1,333 @@
+        }));
+      },
+    },
+    registrants: {
```
@vellames-turing can we include the `sanitizedArray` helper function here (and anywhere else we might accept an array)?

dannyroosevelt commented on components/zoom_admin/actions/delete-webinar-panelist/delete-webinar-panelist.mjs:
```diff
@@ -0,0 +1,39 @@
+        "webinar",
+      ],
+    },
+    panelist: {
```
@vellames-turing  I get duplicated panelists listed here: 
![image](https://user-images.githubusercontent.com/20780753/149569161-a62ac658-9a5c-4dae-b3be-174c3e93d6b4.png)

Is this a bug in the component code or within async options code?

vellames-turing replied:
Fixed!

dannyroosevelt commented on components/zoom_admin/actions/delete-cloud-recording/delete-cloud-recording.mjs:
```diff
@@ -0,0 +1,45 @@
+  type: "action",
+  props: {
+    zoomAdmin,
+    cloudRecording: {
```
@vellames-turing I also get duplicated items listed in the async options dropdown here

vellames-turing replied:
Hey @dannyroosevelt, looks normal to me, when you save some record, dropbox creates two registers, MP4 and M4A, as far as I know, M4A is just for audio.
![Screen Shot 2022-01-20 at 2 19 05 PM](https://user-images.githubusercontent.com/91911778/150389216-6b47c972-6b53-47bf-96ac-0279f2b41612.png)

dannyroosevelt replied:
Ah good call, thanks. So actually shouldn't I see more recordings in this list? 
![image](https://user-images.githubusercontent.com/20780753/150391644-4e06559b-c58e-41de-ae8f-5b0cb5abc639.png)
![image](https://user-images.githubusercontent.com/20780753/150391591-51e1e3d2-663a-4620-8bf7-107d94ccc20d.png)

vellames-turing replied:
Indeed, I will take a look

dannyroosevelt commented:
- [x]  End Meeting | End a meeting for a user
- [x]  Delete Meeting | Remove a meeting by ID
- [x]  List Meeting Registrants | List all registrants of a meeting
- [x]  Update Meeting Registrant Status | Update registrant status for a meeting
- [x]  Get Webinar | Retrieve the details of a webinar
- [x]  List Webinars | List all webinars of a user
- [x]  Update Webinar | Update the details of a webinar
- [x]  Delete Webinar | Remove a webinar for a user
- [x]  List Webinar Registrants | List all registrants of a webinar
- [x]  Update Webinar Registrant Status | Update registrant status for a webinar
- [x]  Add Webinar Panelists | Register multiple panelists for a webinar
- [ ]  Remove Webinar Panelist | Remove a single panelist from a webinar
-- comments added 
- [x]  List Cloud Recordings | Search cloud recordings from a meeting or webinar
- [ ]  Delete a Cloud Recording | Remove a recording from a meeting or webinar
-- comments added

vellames-turing commented:
Hey @dannyroosevelt. I finished the refactor.

- The `withLabel` property was added,
- Some properties was refactored to correctly return the values without the label inside the data structure
- Added more detailed summary messages
- Removed the commented properties
- Removed unused data from HTTP Request due the removal of some peroperties
- Lint problems caused by properties removal fixeds

Also, I retested all the components and looks good

Unfortunately I have two bad news:

- I was not able to figure out why the list of cloud recordings is not showing the records on the API return as we can see on zoom web interface. I tested different params, I even was able to return the trash records, but I can't fetch all the data like in web interface. All the new records that I tested was being returned in API, but the other ones, I really not figure out why are not being returned :(
- The second point is the action `update_meeting_registrant_status`. Zoom is returning an error saying that I am not sending a valid JSON. I read the documentation and I am sure that I am sending it the correct way. We have a similar action with the webinar, with the exatcly data structure working. Maybe I am blind due to looking so many times for zoom docs. 

How should we proceed with this two cases?
Thank you :)

dannyroosevelt commented:
Nice, thanks Cassiano! 

- Cloud recordings issue: this is fine, we've spent enough time on this one, let's publish it.
- `update_meeting_registrant_status` -- let's just not publish this action for now.

So can you merge everything other than that action asap?

vellames-turing commented:
Merged @dannyroosevelt, Thank you!
